viewing_party/party.py

```diff
@@ -0,0 +1,253 @@
+        else: 
+            for movie_dict in user_watched_list:
+                for movie in movie_dict.values():
+                    user_watched_set.add(movie)
+            for friends_movie_dicts in friends_watched_list:
+                for title in friends_movie_dicts.values():
+                    friends_watched_set.add(title)
+
+    set_difference = friends_watched_set - user_watched_set
+    unique_watched_list = list(set_difference)
+
+    friends_unique_watched_list = []
+    for unique_movies in unique_watched_list:
+        list_of_dicts = {"title": unique_movies}
+        friends_unique_watched_list.append(list_of_dicts)
+
+    return friends_unique_watched_list
+
+
+# Wave 4 **********************************************
+
+# 1
+
+def get_available_recs(user_data):
+    user_movies = []
+    for movie in user_data["watched"]:
+        user_movies.append(movie["title"])
+
+    recommended_movies = []
+    for friends_watched in user_data["friends"]:
+        for friend_dicts in friends_watched["watched"]:
+            if (
+                friend_dicts["title"] not in user_movies
+                and friend_dicts["host"] in user_data["subscriptions"]
+                and friend_dicts not in recommended_movies
+            ):
+                recommended_movies.append(friend_dicts)
+    
+    return recommended_movies
+
+
+# Wave 5 **********************************************
+
+# 1
+
+def get_new_rec_by_genre(user_data):
+    import statistics
+    from statistics import mode
+
+    favorite_genres = []
+    for watched in user_data["watched"]:
+        favorite_genres.append(watched["genre"])
+    
+    if len(favorite_genres) == 0:
+        return []
+    else:
+        favorite_genre = mode(favorite_genres)
+
+    user_movies = []
+    for movie in user_data["watched"]:
+        user_movies.append(movie["title"])
+
+    recommended_movies = []
+    for friends_watched in user_data["friends"]:
+        for friend_dicts in friends_watched["watched"]:
+            if (
+                friend_dicts["title"] not in user_movies
+                and friend_dicts["genre"] == favorite_genre
+                and friend_dicts not in recommended_movies
+            ):
+                recommended_movies.append(friend_dicts)
+
+    return recommended_movies
+
+
+# 2
+
+
+def get_rec_from_favorites(user_data):
+
+    favorites = []
+    for faves in user_data["favorites"]:
+        favorites.append(faves["title"])
+    
+    if len(favorites) == 0:
+        return []
+
+    favorites_set = set(favorites)
+    friends_watched_set= set()
+    for friends_watched in user_data["friends"]:
+        for friend_list in friends_watched["watched"]:
+            friends_watched_set.add(friend_list["title"])
+    
+    recommendation = favorites_set - friends_watched_set
+    rec_list = list(recommendation)
+
+    for rec in rec_list:
+        recommended_dict = {"title": rec}
+        recommended_movies = [recommended_dict]
```

Comment:
This is overwriting the `recommended_movies` with a new list with one movie every time through the loop. This only appears to work because the tests are looking for either 0 or 1 result. We shouldn't assume this will always be the case.